### ISSUE TITLE
end-to-end examples should all have criticism

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#181]
We're being a bit lazy in several end-to-end tutorials where we simply posit data, model, then run inference. We should always do some form of criticism to check that the model has even learned anything. The easiest can be point-based evaluation, and with more effort, pretty pictures and/or plots of graphical checks.

**Comment by zzz2010** [comment]
strongly agree with that. For example, I have no idea how to get the posterior distribution of the latent variable p for "beta_bernoulli" example , after "inference.run()".   

It will be great if someone can add that to the criticism section.

Thanks.

**Comment by junpenglao** [comment]
Second that.
@zzz2010 Not sure if it is the correct way but I can get the posterior mean and std of `zs['p']` using:
```
sess = ed.get_session()
inference = ed.MFVI({'p': qp}, data, model)
inference.run(n_iter=500, n_print=50)
mu_post, std_post = sess.run([qp.mean(),qp.std()])
```

**Comment by bhargavvader** [comment]
Will make it a point to do criticism via graphs for whatever i can while making the ipython notebook adaptations!

**Comment by ruzihao** [comment]
@zzz2010 Hi, have you figured out how to get the posterior distribution of the latent variables? Thanks!

**Comment by dustinvtran** [comment]
in that code snippet, the posterior distribution is `qp`. one strength of Edward's design of explicitly defining your posterior approximation (even in MCMC and exact inference) is that the posterior is always available to you.

**Comment by ruzihao** [comment]
@dustinvtran thanks. if i would like to get the point estimate of `qp`, shall i just do `sess.run(qp.mean())` as shown in the code snippet?

**Comment by dustinvtran** [comment]
yep, if you want a point estimate, the posterior mean is generally a good choice.

**Comment by ruzihao** [comment]
@dustinvtran thanks for your prompt reply! However, i do have another question regarding this. i was implementing LDA model and was able to get the posterior mean of certain latent variable with no doubt. 

The problem is, whichever the **n_iter** I specified in inference.run(), the model gives me the **exact same** posterior mean of the latent variable. 

Usually, with more iterations to run, we should have a finer result (at least a **different** one), right? Any thoughts about this?

**Comment by dustinvtran** [comment]
can you ask in another github issue? it could help to keep the issues consistent.

**Comment by ruzihao** [comment]
@dustinvtran Sure!
